torchao/quantization/qat/linear.py

```diff
@@ -200,6 +250,7 @@ class Int8DynActInt4WeightQATLinear(FakeQuantizedLinear):
         precision: precision of weights
         scales_precision: precision of per group scales and zero points
     """
+    _QUANTIZER_CLS = Int8DynActInt4WeightQATQuantizer
```

Comment:
this is a bit weird, can the `activation_config` and `weight_config` be passed from caller in `__init__`?

Reply:
and the args like `groupsize`, `precision`, `scales_precision` should be derived from the config I feel, otherwise  it will be error prone since there will be mismatches between `Int8DynActInt4WeightQATLinear` and `Int8DynActInt4WeightQATQuantizer`

Reply:
I think that's also weird, because `Int8DynActInt4WeightQATLinear` should always have the same fake quantize configs. If users pass in other random configs then this linear will start doing unexpected things

Reply:
Also this is just for the legacy code path. Eventually we want users to use `FakeQuantizedLinear` directly for everything, but for now we need to continue to expose this class for BC (many users are using it today)

Reply:
OK in that case, should we hard code the configs in `Int8DynActInt4WeightQATLinear`

Reply:
Ok, but we'll have to duplicate the configs in that case (once in quantizer and once in linear class). Is that OK?

Reply:
hmmm, if you want to avoid mismatch, maybe you could refactor `_get_activation_fake_quantize_config` and `_get_weight_fake_quantize_config` to standalone helper functions so you don't need to set `_QUANTIZER_CLS` here, or considering you'll be deprecating `Int8DynActInt4WeightQATQuantizer`, it might be fine to duplicate as well

Reply:
Ok, separated into standalone helper functions